### ISSUE TITLE
💅 useTemplate problem

### ISSUE THREAD

**Issue by Jack-Works** [biomejs/biome#2580]
### Environment information

```bash
CLI:
  Version:                      1.7.1
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "xterm"
  JS_RUNTIME_VERSION:           "v21.7.3"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "pnpm/9.0.5"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 true

Linter:
  Recommended:                  true
  All:                          false
  Rules:                        complexity/noMultipleSpacesInRegularExpressionLiterals = "off"

Workspace:
  Open Documents:               0
```

### Rule name

useTemplate

### Playground link

https://biomejs.dev/playground/?code=JwBgAGAAYAB0AHMAXABuACcAIAArACAAeAAgACsAIAAnAFwAbgBgAGAAYAAnADsACgA%3D

### Expected result

No error because the current code is readable enough. Instead, biome suggests a syntax invalid fix, and the readability also gets worse.

```
  ✖ Template literals are preferred over string concatenation.
  
  > 1 │ '```ts\n' + x + '\n```';
      │ ^^^^^^^^^^^^^^^^^^^^^^^
    2 │ 
  
  ℹ Unsafe fix: Use a template literal.
  
    1   │ - '```ts\n'·+·x·+·'\n```';
      1 │ + `\`\`\`ts\n${x}\n`\`\``;
    2 2 │   
```

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct
